presto-hive/src/main/java/com/facebook/presto/twitter/hive/CachingHiveMetastoreRetryConfigurable.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive;
+
+import com.facebook.presto.hive.ForHiveMetastore;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveCluster;
+import com.facebook.presto.hive.RetryDriver;
+import com.facebook.presto.hive.metastore.CachingHiveMetastore;
+import com.facebook.presto.spi.PrestoException;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+
+import java.util.concurrent.ExecutorService;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@ThreadSafe
+public class CachingHiveMetastoreRetryConfigurable
```

Comment:
please add javadocs explaining why this class is necessary. if this is just temporary for troubleshooting, let's say that.